pkg/ucp/api/v20231001preview/aws_credential_conversion.go

```diff
@@ -82,13 +82,47 @@ func (cr *AwsCredentialResource) getDataModelCredentialProperties() (*datamodel.
 		}
 
 		return &datamodel.AWSCredentialResourceProperties{
-			Kind: datamodel.AWSCredentialKind,
+			Kind: datamodel.AWSAccessKeyCredentialKind,
 			AWSCredential: &datamodel.AWSCredentialProperties{
-				AccessKeyID:     to.String(p.AccessKeyID),
-				SecretAccessKey: to.String(p.SecretAccessKey),
+				Kind: datamodel.AWSAccessKeyCredentialKind,
+				AccessKeyCredential: &datamodel.AWSAccessKeyCredentialProperties{
+					AccessKeyID:     to.String(p.AccessKeyID),
+					SecretAccessKey: to.String(p.SecretAccessKey),
+				},
 			},
 			Storage: storage,
 		}, nil
+	case *AwsIRSACredentialProperties:
+		var storage *datamodel.CredentialStorageProperties
+
+		switch c := p.Storage.(type) {
+		case *InternalCredentialStorageProperties:
+			if c.Kind == nil {
+				return nil, &v1.ErrModelConversion{PropertyName: "$.properties", ValidValue: "not nil"}
+			}
+			storage = &datamodel.CredentialStorageProperties{
+				Kind: datamodel.InternalStorageKind,
+				InternalCredential: &datamodel.InternalCredentialStorageProperties{
+					SecretName: to.String(c.SecretName),
+				},
+			}
+		case nil:
+			return nil, &v1.ErrModelConversion{PropertyName: "$.properties.storage", ValidValue: "not nil"}
```

Comment:
Instead of `not nil` should we have a proper error message?

Reply:
I think we have this since storage is a structure like below: 
        "storage":{
            "kind":"Internal"
        }

We seem to return the "not nil" error message for all structures, for example 

```
case *InternalCredentialStorageProperties:
			if c.Kind == nil {
				return nil, &v1.ErrModelConversion{PropertyName: "$.properties", ValidValue: "not nil"}
			}
```